src/App.tsx

```diff
@@ -14,11 +15,114 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+interface State {
+  selecedGoods: string[],
+}
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    selecedGoods: ['Jam'],
+  };
+
+  selectGood = (good: string) => {
+    this.setState((prevState) => (
+      { selecedGoods: [...prevState.selecedGoods, good] }
+    ));
+  };
+
+  removeGood = (good: string) => {
+    this.setState((prevState) => ({
+      selecedGoods: prevState.selecedGoods
+        .filter(remaining => remaining !== good),
+    }));
+  };
+
+  clear = () => {
+    this.setState({ selecedGoods: [] });
+  };
+
+  buttonAction = (isSelected: boolean, good: string) => {
+    return isSelected
+      ? this.removeGood(good)
+      : this.selectGood(good);
+  };
+
+  createMessage = () => {
+    const { selecedGoods } = this.state;
+
+    switch (selecedGoods.length) {
+      case 0:
+        return 'No goods selected';
+
+      case 1:
+        return `${selecedGoods[0]} is selected`;
+
+      case 2:
+        return `${selecedGoods[0]} and ${selecedGoods[1]} are selected`;
+
+      default:
+        return `${selecedGoods.slice(0, -1).join(', ')} and ${selecedGoods[selecedGoods.length - 1]} are selected`;
```

Comment:
`.at` will still be better
```suggestion
        return `${selecedGoods.slice(0, -1).join(', ')} and ${selecedGoods.at(-1)} are selected`;
```